javascript/react/props-iii/clone-elements-in-react.md

```diff
@@ -26,42 +26,93 @@ aspects:
 
 ---
 
-# Clone Elements in **React**
+# Clone Elements in React
 
 ---
 ## Content
 
-**React** provides different ways to clone elements. The `React.cloneElement` method, unlike `React.addons.cloneWithProps` can clone an element whilst keeping the `key` and `ref` of the original one.
+**React** provides different ways to clone elements. The `React.cloneElement` method can clone an element whilst keeping the `key` and `ref` of the original one.
 
 How to use:
-```javascript
-ReactElement cloneElement(
-  ReactElement element,
-  [object props],
-  [children ...]
+```jsx
+React.cloneElement(
+  element,
+  [props],
+  [...children]
 );
 ```
 
 A new `ReactElement` will be returned using the provided `element` as the starting point. The old `props` will be merged in shallowly with the new specified ones. Furthermore, new `children` will replace old ones.
```

Comment:
```suggestion
A new element will be returned using the provided element as the starting point. The old `props` will be merged in shallowly with the new specified ones. Furthermore, new `children` will replace old ones.
```

Reply:
We can add 2 footnotes:

1) Explaining the element similarly to [here](https://reactjs.org/docs/glossary.html#elements)
2) Explaining "shallow merge" with something along the lines of:

> A merge of two objects in JavaScript produces a new object by combining their properties. Properties that only exists in one of the two objects are just copied into the result. If a property exists in both, the value of that property from the second object will overwrite the value of the first. A"shallow" merge means that we only compare/merge top-level properties. Merging all levels of properties between two objects is called a *deep merge*.

Reply:
Added the following footnotes:

[1: element]
The React element represents the building block of any React application. Although one might confuse elements with components, they are not the same. An element is used to describe what you want to see on the screen. Here's an example:
```jsx
const element = <h1>Enki is cool!</h1>;
```

[2: shallow merge]
A merge of two objects in JavaScript produces a new object by combining their properties. Properties that only exists in one of the two objects are just copied into the result. If a property exists in both, the value of that property from the second object will overwrite the value of the first. A"shallow" merge means that we only compare/merge top-level properties. Merging all levels of properties between two objects is called a *deep merge*.